### PR TITLE
Sink 'docker login' step to right before image push and other small cleanups

### PR DESCRIPTION
**Pull request by nathanchance**
As suggested by @compnerd on #19. This includes a few other small clean ups that I noticed but those can be factored out if necessary.

### CODE REVIEW

File: .github/workflows/llvm-project-epoch-one.yml [hunk 1]
```diff
@@ -10,15 +10,10 @@ jobs:
     steps:
       - name: Checkout repo
         uses: actions/checkout@v3
+
```
Comment (compnerd) [inline .github/workflows/llvm-project-epoch-one.yml:13]: Thank you for the whitespace!

File: .github/workflows/llvm-project-epoch-one.yml [hunk 2]
```diff
@@ -28,8 +23,18 @@ jobs:
...
         run: bash ci/test-clang.sh
+
+      - name: Login to ghcr.io
+        if: ${{ github.repository_owner == 'ClangBuiltLinux' && github.event_name != 'pull_request' && github.ref_name == 'main' }}
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
```
Comment (compnerd) [inline .github/workflows/llvm-project-epoch-one.yml:37]: Does it make sense to pull this out into a local composite action (combining the login/push)?

**Review by compnerd — APPROVED** [review]
This definitely feels more organized, thanks!

File: .github/workflows/llvm-project-epoch-one.yml (hunk 2 above)
Comment (nathanchance) [reply .github/workflows/llvm-project-epoch-one.yml:37]: Yes, I intend to take a look at sharing more between workflows after the weekend, as I am already sad at how much duplication we have around here :( although I am not sure I want to start that process until I see how the workflows are going to evolve with each additional stage but I am sure there is some work that can be done before that.

Comment (nathanchance) [reply .github/workflows/llvm-project-epoch-one.yml:37]: I wish we could use [a reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows) for this but the one gotcha is that we cannot use a `strategy` property, which is going to be an issue if we want to wire up arm64 support (which I already have working).

A composite action seems like it cannot be local and it seems like a big hammer to add a whole new repository for two steps...

Comment (compnerd) [reply .github/workflows/llvm-project-epoch-one.yml:37]: I'm not sure I am following; I think that we should be able to do that with some bit of specialization.  I can try to put up an example of what I mean once this is merged.

**Pull request merged**